File: src/main/java/br/com/unopay/api/market/model/ContractorBonusBilling.java
package br.com.unopay.api.market.model;

import br.com.unopay.api.bacen.model.Contractor;
import br.com.unopay.api.model.Updatable;
import lombok.Data;
import org.hibernate.annotations.GenericGenerator;
import org.springframework.data.annotation.Id;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.validation.Valid;
import java.io.Serializable;

@Data
@Entity
@Table(name = "contractor_bonus_billing")
public class ContractorBonusBilling implements Serializable, Updatable {

    private static final long serialVersionUID = 2732233885546623588L;

    public ContractorBonusBilling() {}

    @Id
    @Column(name = "id")
    @GeneratedValue(generator = "system-uuid")
    @GenericGenerator(name = "system-uuid", strategy = "uuid2")
    private String id;

    @Valid
    @ManyToOne
    @JoinColumn(name = "bonus_billing_id")
    private BonusBilling bonusBilling;


    @Valid
    @ManyToOne
    @JoinColumn(name = "contractor_bonus_id")
    private Contractor contractor;
}

File: src/main/scala/br/com/unopay/api/market/model/ContractorBonusBilling.scala
package br.com.unopay.api.market.model

import java.io.Serializable
import javax.persistence._
import javax.validation.Valid

import lombok.Data
import org.hibernate.annotations.GenericGenerator

import scala.beans.BeanProperty


@Data
@Entity
@Table(name = "contractor_bonus_billing")
class ContractorBonusBilling extends Serializable {

    @Id
    @BeanProperty
    @Column(name = "id")
    @GeneratedValue(generator = "system-uuid")
    @GenericGenerator(name = "system-uuid", strategy = "uuid2")
    var id: String = _

    @Valid
    @BeanProperty
    @ManyToOne
    @JoinColumn(name = "contractor_bonus_id")
    var contractorBonus: ContractorBonus = _

    @Valid
    @BeanProperty
    @ManyToOne
    @JoinColumn(name = "bonus_billing_id")
    var bonusBilling: BonusBilling = _
}
